tasks/rulegen/src/main.rs

```diff
@@ -175,6 +175,39 @@ fn format_code_snippet(code: &str) -> String {
     format!("r#\"{}\"#", code.replace("\\\"", "\""))
 }
 
+fn dedent(s: &str) -> String {
+    let lines: Vec<&str> = s.lines().collect();
+    if lines.is_empty() {
+        return String::new();
+    }
+    // find the min whitespace count
+    let min_indent = lines
+        .iter()
```

Comment:
I don't think we need to `collect` here, since we are only ever using `.iter()` with it anyway. This will save some memory because we don't need to allocate space for all lines.

```suggestion
    // find the min whitespace count
    let min_indent = s
        .lines()
        .iter()
```